### DISCUSSION TITLE
Disconnection Timeout

### DISCUSSION THREAD

**Discussion by francois-fiveco** [bluez/bluez#1032] [Q&A]
Hi,
Sometimes, the HCI command "Disconnect" ends with "LMP Response Timeout / LL Response Timout" but bluetoothctl considers that it is always connected. 
Is there a way to make bluetoothctl understand that the device is disconnected?

Btmon Disconnection succeeded :
```
bluetoothd[966]: @ MGMT Comm..d (0x0014) plen 7  {0x0001} [hci0] 17:00:25.328512
        LE Address: C7:DA:7A:BF:5C:4C (Static)
< HCI Command: Disconnect (0x01|0x0006) plen 3      #666 [hci0] 17:00:25.328747
        Handle: 64 Address: C7:DA:7A:BF:5C:4C (Static)
        Reason: Remote User Terminated Connection (0x13)
> HCI Event: Command Status (0x0f) plen 4           #667 [hci0] 17:00:25.329138
      Disconnect (0x01|0x0006) ncmd 1
        Status: Success (0x00)
> HCI Event: Disconnect Complete (0x05) plen 4      #668 [hci0] 17:00:25.335385
        Status: Success (0x00)
        Handle: 64 Address: C7:DA:7A:BF:5C:4C (Static)
        Reason: Connection Terminated By Local Host (0x16)
@ MGMT Event: Command Compl.. (0x0001) plen 10  {0x0001} [hci0] 17:00:25.335418
      Disconnect (0x0014) plen 7
        Status: Success (0x00)
        LE Address: C7:DA:7A:BF:5C:4C (Static)
< HCI Command: LE Remove De.. (0x08|0x0012) plen 7  #669 [hci0] 17:00:26.933350
        Address type: Random (0x01)
        Address: C7:DA:7A:BF:5C:4C (Static)
> HCI Event: Command Complete (0x0e) plen 4         #670 [hci0] 17:00:26.933698
      LE Remove Device From Accept List (0x08|0x0012) ncmd 1
        Status: Success (0x00)
```

Disconnection Failed :
```
bluetoothd[966]: @ MGMT Comm..d (0x0014) plen 7  {0x0001} [hci0] 09:00:39.328509
        LE Address: FE:9F:53:8C:16:AE (Static)
< HCI Command: Disconnect (0x01|0x0006) plen 3     #1977 [hci0] 09:00:39.328596
        Handle: 64 Address: FE:9F:53:8C:16:AE (Static)
        Reason: Remote User Terminated Connection (0x13)
> HCI Event: Command Status (0x0f) plen 4          #1978 [hci0] 09:00:39.328967
      Disconnect (0x01|0x0006) ncmd 1
        Status: Success (0x00)
> HCI Event: Disconnect Complete (0x05) plen 4     #1979 [hci0] 09:00:44.317274
        Status: Success (0x00)
        Handle: 64 Address: FE:9F:53:8C:16:AE (Static)
        Reason: LMP Response Timeout / LL Response Timeout (0x22)
< HCI Command: LE Remove D.. (0x08|0x0012) plen 7  #1980 [hci0] 09:00:48.179157
        Address type: Random (0x01)
        Address: FE:9F:53:8C:16:AE (Static)
> HCI Event: Command Complete (0x0e) plen 4        #1981 [hci0] 09:00:48.179491
      LE Remove Device From Accept List (0x08|0x0012) ncmd 1
        Status: Success (0x00) 
```

**Comment by Vudentz** [comment]
> Hi, Sometimes, the HCI command "Disconnect" ends with "LMP Response Timeout / LL Response Timout" but bluetoothctl considers that it is always connected. Is there a way to make bluetoothctl understand that the device is disconnected?
> 
> Btmon Disconnection succeeded :
> 
> ```
> bluetoothd[966]: @ MGMT Comm..d (0x0014) plen 7  {0x0001} [hci0] 17:00:25.328512
>         LE Address: C7:DA:7A:BF:5C:4C (Static)
> < HCI Command: Disconnect (0x01|0x0006) plen 3      #666 [hci0] 17:00:25.328747
>         Handle: 64 Address: C7:DA:7A:BF:5C:4C (Static)
>         Reason: Remote User Terminated Connection (0x13)
> > HCI Event: Command Status (0x0f) plen 4           #667 [hci0] 17:00:25.329138
>       Disconnect (0x01|0x0006) ncmd 1
>         Status: Success (0x00)
> > HCI Event: Disconnect Complete (0x05) plen 4      #668 [hci0] 17:00:25.335385
>         Status: Success (0x00)
>         Handle: 64 Address: C7:DA:7A:BF:5C:4C (Static)
>         Reason: Connection Terminated By Local Host (0x16)
> @ MGMT Event: Command Compl.. (0x0001) plen 10  {0x0001} [hci0] 17:00:25.335418
>       Disconnect (0x0014) plen 7
>         Status: Success (0x00)
>         LE Address: C7:DA:7A:BF:5C:4C (Static)
> < HCI Command: LE Remove De.. (0x08|0x0012) plen 7  #669 [hci0] 17:00:26.933350
>         Address type: Random (0x01)
>         Address: C7:DA:7A:BF:5C:4C (Static)
> > HCI Event: Command Complete (0x0e) plen 4         #670 [hci0] 17:00:26.933698
>       LE Remove Device From Accept List (0x08|0x0012) ncmd 1
>         Status: Success (0x00)
> ```
> 
> Disconnection Failed :
> 
> ```
> bluetoothd[966]: @ MGMT Comm..d (0x0014) plen 7  {0x0001} [hci0] 09:00:39.328509
>         LE Address: FE:9F:53:8C:16:AE (Static)
> < HCI Command: Disconnect (0x01|0x0006) plen 3     #1977 [hci0] 09:00:39.328596
>         Handle: 64 Address: FE:9F:53:8C:16:AE (Static)
>         Reason: Remote User Terminated Connection (0x13)
> > HCI Event: Command Status (0x0f) plen 4          #1978 [hci0] 09:00:39.328967
>       Disconnect (0x01|0x0006) ncmd 1
>         Status: Success (0x00)
> > HCI Event: Disconnect Complete (0x05) plen 4     #1979 [hci0] 09:00:44.317274
>         Status: Success (0x00)
>         Handle: 64 Address: FE:9F:53:8C:16:AE (Static)
>         Reason: LMP Response Timeout / LL Response Timeout (0x22)
> < HCI Command: LE Remove D.. (0x08|0x0012) plen 7  #1980 [hci0] 09:00:48.179157
>         Address type: Random (0x01)
>         Address: FE:9F:53:8C:16:AE (Static)
> > HCI Event: Command Complete (0x0e) plen 4        #1981 [hci0] 09:00:48.179491
>       LE Remove Device From Accept List (0x08|0x0012) ncmd 1
>         Status: Success (0x00) 
> ```

I don't see anything wrong here, well except that the Reason is different and perhaps the controller still thinks it is connected, or you don't see it being disconnected at upper layers?

**Reply by Vishal8523** [reply]
any workaround that one can do ??
